pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint.go

```diff
@@ -49,35 +51,31 @@ var toolsMount = corev1.VolumeMount{
 }
 
 // Cache is a simple caching mechanism allowing for caching the results of
-// getting the Entrypoint of a container image from a remote registry. It
-// is synchronized via a mutex so that we can share a single Cache across
-// each worker thread that the reconciler is running. The mutex is necessary
-// due to the possibility of a panic if two workers were to attempt to read and
-// write to the internal map at the same time.
+// getting the Entrypoint of a container image from a remote registry. The
+// internal lru cache is thread-safe.
 type Cache struct {
-	mtx   sync.RWMutex
-	cache map[string][]string
+	lru *lru.Cache
 }
 
 // NewCache is a simple helper function that returns a pointer to a Cache that
-// has had the internal cache map initialized.
+// has had the internal fixed-sized lru cache initialized.
 func NewCache() *Cache {
-	return &Cache{
-		cache: make(map[string][]string),
+	lru, err := lru.New(defaultEntryPointCacheSize)
+	if err != nil {
+		panic(err)
```

Comment:
We shouldn't panic here, we should return the error from `NewCache` and update callers to handle it.

Reply:
THX for review. Has fixed code as you requested. @ImJasonH